test/api_integration/apis/saved_queries/saved_queries.js

```diff
@@ -150,5 +194,45 @@ export default function ({ getService }) {
         .delete(`${SAVED_QUERY_BASE_URL}/invalid_id`)
         .set(ELASTIC_HTTP_VERSION_HEADER, '1')
         .expect(404));
+
+    it('should return isDuplicate = true for _is_duplicate_title check with a duplicate title', () =>
+      supertest
+        .post(`${SAVED_QUERY_BASE_URL}/_is_duplicate_title`)
+        .set(ELASTIC_HTTP_VERSION_HEADER, '1')
+        .send({ title: 'my title' })
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.isDuplicate).to.be(true);
+        }));
+
+    it('should return isDuplicate = false for _is_duplicate_title check with a duplicate title and matching ID', () =>
+      supertest
+        .post(`${SAVED_QUERY_BASE_URL}/_create`)
+        .set(ELASTIC_HTTP_VERSION_HEADER, '1')
+        .send({
+          ...mockSavedQuery,
+          title: 'my title 7',
+        })
+        .expect(200)
+        .then(({ body }) =>
+          supertest
+            .post(`${SAVED_QUERY_BASE_URL}/_is_duplicate_title`)
+            .set(ELASTIC_HTTP_VERSION_HEADER, '1')
+            .send({ title: 'my title 7', id: body.id })
+            .expect(200)
+            .then(({ body }) => {
+              expect(body.isDuplicate).to.be(false);
+            })
+        ));
+
+    it('should return isDuplicate = false for _is_duplicate_title check with a unique title', () =>
+      supertest
+        .post(`${SAVED_QUERY_BASE_URL}/_is_duplicate_title`)
+        .set(ELASTIC_HTTP_VERSION_HEADER, '1')
+        .send({ title: 'my unique title' })
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.isDuplicate).to.be(false);
+        }));
```

Comment:
Makes sense to me. I ended up adding tests for searching, pagination with and without a search, count requests, and a few other scenarios, as well as improving the existing tests. I also made a few other changes that I think improves the API tests:
- Converted the test file to TypeScript to get autocomplete, etc.
- Started clearing the queries between tests so we don't need to work around the existing queries.
- Added some utility functions to make writing new tests quicker and easier to read.
- Reorganized the tests by endpoint since there are quite a few of them now.